dlt/destinations/job_client_impl.py

```diff
@@ -523,36 +520,137 @@ def _from_db_type(
         pass
 
     def get_stored_schema(self, schema_name: str = None) -> StorageSchemaInfo:
-        name = self.sql_client.make_qualified_table_name(self.schema.version_table_name)
-        c_schema_name, c_inserted_at = self._norm_and_escape_columns("schema_name", "inserted_at")
-        if not schema_name:
-            query = (
-                f"SELECT {self.version_table_schema_columns} FROM {name}"
-                f" ORDER BY {c_inserted_at} DESC;"
-            )
-            return self._row_to_schema_info(query)
-        else:
-            query = (
-                f"SELECT {self.version_table_schema_columns} FROM {name} WHERE {c_schema_name} = %s"
-                f" ORDER BY {c_inserted_at} DESC;"
+        table_name = self.schema.version_table_name
+        dataset_name = self.sql_client.dataset_name
+        catalog_name = self.sql_client.catalog_name(escape=False)
+
+        if self.config.destination_type == "clickhouse":
+            dataset_table_separator = self.config.dataset_table_separator  # type: ignore[attr-defined]
+            table_name = dataset_table_separator.join([dataset_name, table_name])
+            dataset_name = catalog_name
+            catalog_name = None
+
+        c_schema_name, c_inserted_at = [
+            self.schema.naming.normalize_path(p) for p in ["schema_name", "inserted_at"]
+        ]
+
+        table_expr = sqlglot.exp.Table(
+            this=sqlglot.exp.to_identifier(table_name, quoted=True),
+            db=sqlglot.exp.to_identifier(dataset_name, quoted=True),
```

Comment:
If we use this new function to normalize queries (explained below), we will not have to add db and catalog everywhere.